sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosTableSchemaInferrer.scala

```diff
@@ -79,10 +81,15 @@ private object CosmosTableSchemaInferrer
         case _ => cosmosReadConfig.inferSchemaQuery.get
       }
 
-      val queryObservable =
+      val totalResults = new AtomicLong(0)
+      val pagedFluxResponse =
         sourceContainer.queryItems(queryText, queryOptions, classOf[ObjectNode])
 
-      val feedResponseList = queryObservable.byPage.collectList.block
+      val feedResponseList = pagedFluxResponse
+        .byPage
+        .takeUntil(page => totalResults.addAndGet(page.getResults.size) >= cosmosReadConfig.inferSchemaSamplingSize)
+        .collectList
+        .block
```

Comment:
@ealsur there is easier way to do this and you don't need the AtomicLong, etc

see this: https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#take-long-

```scala
pagedFluxResponse.take(cosmosReadConfig.inferSchemaSamplingSize)
     .toList
     .block
```

should do the trick.

Reply:
Done